test/unit/io/alignment_file/alignment_file_format_test_template.hpp

```diff
@@ -202,196 +211,109 @@ TYPED_TEST_P(alignment_file_read, read_in_all_data)
     this->tag_dicts[1]["bI"_tag] = std::vector<uint32_t>{294967296u};
     this->tag_dicts[1]["bf"_tag] = std::vector<float>{3.5f, 0.1f, 43.8f};
 
-    dna5_vector seq;
-    std::string id;
-    std::vector<phred42> qual;
-    int32_t offset;
-    std::optional<int32_t> ref_id_in;
-    std::optional<int32_t> ref_offset;
-    std::pair<std::vector<gapped<dna5>>, std::vector<gapped<dna5>>> alignment;
-    uint16_t flag;
-    uint8_t mapq;
-    std::tuple<std::optional<int32_t>, std::optional<int32_t>, int32_t> mate;
-    sam_tag_dictionary tag_dict;
-
-    for (size_t i = 0; i < 3; ++i)
+    size_t i{0};
+    for (auto & rec : fin)
     {
-        ASSERT_NO_THROW(format.read(istream, input_options, this->ref_sequences, this->header, seq, qual, id, offset,
```

Comment:
I disagree because in this case the change reduces the LOC by 50% and with your way I would not reduce the number of *added* lines but just the number of *removed* lines.